codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -175,11 +179,15 @@ public boolean goAwaySent() {
 
     @Override
     public void goAwaySent(int lastKnownStream, long errorCode, ByteBuf debugData) {
-        boolean alreadyNotified = goAwaySent();
         remoteEndpoint.lastKnownStream(lastKnownStream);
-        if (!alreadyNotified) {
-            for (Listener listener : listeners) {
-                listener.onGoAwaySent(lastKnownStream, errorCode, debugData);
+        for (Listener listener : listeners) {
+            listener.onGoAwaySent(lastKnownStream, errorCode, debugData);
+        }
+
+        Http2Stream[] streams = new Http2Stream[numActiveStreams()];
+        for (Http2Stream stream : activeStreams().toArray(streams)) {
+            if (stream.id() > lastKnownStream && remoteEndpoint.createdStreamId(stream.id())) {
```

Comment:
`stream.id()` may be used multiple times...should we put it in a temporary variable?